content/en/logs/guide/build-custom-reports-using-log-analytics-api.md

```diff
@@ -0,0 +1,667 @@
+       "query":"*"
+           },
+   "group_by":[
+       {
+           "type":"facet",
+           "facet":"@http.useragent_details.os.family",
+           "limit":2,
+           "sort":{
+               "order":"desc",
+               "type":"measure",
+               "aggregation":"cardinality",
+               "metric":"@http.useragent"
+           }
+       },
+       {
+           "type":"facet",
+           "facet":"@http.useragent_details.browser.family",
+           "limit":2,
+           "sort":{
+               "order":"desc",
+               "type":"measure",
+               "aggregation":"cardinality",
+               "metric":"@http.useragent"
+           }
+       }
+   ]
+}
+'
+
+```
+
+**Response:**
+
+```json
+{
+    "meta": {
+        "status": "done",
+        "request_id": "dkt3bGhON0lSOEdCVWFqa3pyUEtNUXxzU0p5RG1qN3MwNk45aExrazFGTTR3",
+        "elapsed": 1299
+    },
+    "data": {
+        "buckets": [
+            {
+                "computes": {
+                    "c3": 534310,
+                    "c2": 29855.686900195342,
+                    "c1": 289880482.9557167,
+                    "c0": 430
+                },
+                "by": {
+                    "@http.useragent_details.browser.family": "Chrome",
+                    "@http.useragent_details.os.family": "Mac OS X"
+                }
+            },
+            {
+                "computes": {
+                    "c3": 47973,
+                    "c2": 25117.50770936209,
+                    "c1": 270379443.2579185,
+                    "c0": 64
+                },
+                "by": {
+                    "@http.useragent_details.browser.family": "Firefox",
+                    "@http.useragent_details.os.family": "Mac OS X"
+                }
+            },
+            {
+                "computes": {
+                    "c3": 901506,
+                    "c2": 9170.975124352715,
+                    "c1": 235075236.08510733,
+                    "c0": 342
+                },
+                "by": {
+                    "@http.useragent_details.browser.family": "Other",
+                    "@http.useragent_details.os.family": "Other"
+                }
+            },
+            {
+                "computes": {
+                    "c3": 2734,
+                    "c2": 953181.3177150192,
+                    "c1": 200800000.00000006,
+                    "c0": 45
+                },
+                "by": {
+                    "@http.useragent_details.browser.family": "Apache-HttpClient",
+                    "@http.useragent_details.os.family": "Other"
+                }
+            }
+        ]
+    }
+}
+
+```
+In the response, `c0` represents the unique count of `useragent`, `c1` represents the `pc90` of metric `duration`, `c2` represents the `avg` of metric `network.bytes_written`, and `c3` represents the total `count` of log events.
+
+**Note:** Paging is only supported if `sort` is `alphabetical`.
```

Comment:
Should we link to the main pagination guide and add an example on analytics there?

Reply:
I will add an example on main pagination guide separately.